### PR TITLE
oauth2-server/id_token: id_token now returned with refresh_token

### LINKED ISSUE #44: oauth2: return id_token when refreshing existing token
The id_token need to be returned when a existing token is refreshed using the refresh token. 

Related with #43

### PR DESCRIPTION
**Pull request by VasilMitovClouway**
Now when token is refreshed, id_token is returned in the response.

Fixes #44

### CODE REVIEW

File: oauth2-server/src/main/java/com/clouway/oauth2/OAuth2Servlet.java [hunk 1]
```diff
@@ -66,7 +67,7 @@ public void init(ServletConfig servletConfig) throws ServletException {
                                                                     config.identityFinder(),
                                                                     new IssueNewTokenActivity(
                                                                             config.tokens(),
-                                                                            new JjwtIdTokenFactory(config.keyStore())
+                                                                            idIdTokenFactory
```
Comment (mgenov) [inline oauth2-server/src/main/java/com/clouway/oauth2/OAuth2Servlet.java]: idId doesn't sound good. 

idTokenFactory

**Review by mgenov — CHANGES_REQUESTED** [review]

File: oauth2-server/src/main/java/com/clouway/oauth2/OAuth2Servlet.java (hunk 1 above)
Comment (VasilMitovClouway) [reply oauth2-server/src/main/java/com/clouway/oauth2/OAuth2Servlet.java]: I have overlooked this naming. Renamed to idTokenFactory

File: oauth2-server/src/main/java/com/clouway/oauth2/RefreshTokenActivity.java [hunk 2]
```diff
@@ -1,21 +1,28 @@
...
  * @author Miroslav Genov (miroslav.genov@clouway.com)
  */
 class RefreshTokenActivity implements ClientActivity {
 
   private final Tokens tokens;
+  private final IdTokenFactory idIdTokenFactory;
+  private final IdentityFinder identityFinder;
 
-  RefreshTokenActivity(Tokens tokens) {
+  RefreshTokenActivity(Tokens tokens, IdTokenFactory idIdTokenFactory, IdentityFinder identityFinder) {
```
Comment (mgenov) [inline oauth2-server/src/main/java/com/clouway/oauth2/RefreshTokenActivity.java]: Still idIdTokenFactory.

**Review by mgenov — CHANGES_REQUESTED** [review]

File: oauth2-server/src/main/java/com/clouway/oauth2/RefreshTokenActivity.java (hunk 2 above)
Comment (VasilMitovClouway) [reply oauth2-server/src/main/java/com/clouway/oauth2/RefreshTokenActivity.java]: Changed. I found a few more places that was named idId and changed them as well

File: oauth2-server/src/test/java/com/clouway/oauth2/RefreshTokenForClientTest.java [hunk 3]
```diff
@@ -28,34 +36,139 @@
...
+                      "::refresh_token::")
+      ));
+
+      oneOf(identityFinder).findIdentity("::identityId::", GrantType.AUTHORIZATION_CODE, anyTime, Collections.<String, String>emptyMap());
+      will(returnValue(Optional.of(identity)));
+
+      oneOf(request).header("Host");
+      will(returnValue("::host::"));
+
+      oneOf(idTokenFactory).create("::host::", client.id, identity, 600L, anyTime);
```
Comment (mgenov) [inline oauth2-server/src/test/java/com/clouway/oauth2/RefreshTokenForClientTest.java:123]: identity is absent why you are trying to generated token for that identity ?

Comment (VasilMitovClouway) [reply oauth2-server/src/test/java/com/clouway/oauth2/RefreshTokenForClientTest.java:123]: I think you are not looking at the right place. identityNotFound test begins at line 138 and at line 160 im returning optional absent and asserting that the body contans "identity was not found"

**Review by mgenov — APPROVED** [review]
LGTM !

**Pull request merged**